pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -264,6 +319,43 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 	return nil
 }
 
+// getDefaultGateway returns the default gateway address with lower priority using route and device
+// statedb tables and the provided default gateway configuration.
+func (r *NeighborReconciler) getDefaultGateway(defaultGateway *v2.DefaultGateway) (string, error) {
+	var defaultRoute netip.Prefix
+	switch defaultGateway.AddressFamily {
+	case "ipv4":
+		defaultRoute = ipv4Default
+	case "ipv6":
+		defaultRoute = ipv6Default
+	default:
+		return "", fmt.Errorf("invalid address family %s", defaultGateway.AddressFamily)
+	}
+	txn := r.DB.ReadTxn()
+	// get routes from statedb route table
+	routes := r.routeTable.All(txn)
```

Comment:
@YutaroHayakawa your suggestion to look for default routes using `RoutePrefixIndex` does not exist so i'm just fetching all the routes here.
https://github.com/cilium/cilium/pull/38767/files#r2045389635

Reply:
i added a TODO for it